db/migrate/20211112035233_add_looking_for_work_to_developers.rb

```diff
@@ -0,0 +1,14 @@
+class AddLookingForWorkToDevelopers < ActiveRecord::Migration[7.0]
+  def up
+    add_column :developers, :looking_for_work, :boolean
+
+    Developer.where("available_on < ?", Date.today + 1.year).update_all(looking_for_work: true)
+    Developer.where(looking_for_work: nil).update_all(looking_for_work: false)
+
+    change_column :developers, :looking_for_work, :boolean, null: false, default: true
+  end
```

Comment:
This _should_ be equivalent, i.e. add a non-null column, with a default, then update the developers that are not _looking for work_:

```suggestion
  def up
    add_column :developers, :looking_for_work, :boolean, null: false, default: true

    Developer.where("available_on >= ?", Date.today + 1.year).update_all(looking_for_work: false)
  end
```

It would be sensible to add an index to this column too e.g:

```
add_column :developers, :looking_for_work, :boolean, null: false, default: true, index: true
```

Reply:
I don't think adding an index to a column whose cardinality is less will give us any improvements, let me research more and get back on this

Reply:
confirming that there are no gains when you add an index, whose cardinality is less.

References:
https://stackoverflow.com/questions/1844991/indexing-boolean-fields
https://stackoverflow.com/a/10524734/1395129

Reply:
Thanks @vamsipavanmahesh that's good to know.

Reply:
I agree that an index should **not** be added, but it does look like there are certain scenarios where an index could be beneficial.

For example, 10,000 developers register on the site and only 10 developers are 'looking for work', apparently, indexing would be more efficient. There's a good explanation here: https://stackoverflow.com/a/43273820

I guess the indexing will be determined by the dataset, which cannot be predicted in advance.

Reply:
Yeah I hope this project won't have such data haha .. but let's see depending on the data and queries may be we can defer this

Reply:
Good call on the code suggestion, @davegudge! I accepted it.

I'm going to hold off on the index for now. Thanks for the research.